prov/util/src/ipc_handle_monitor.c

```diff
@@ -0,0 +1,77 @@
+/*
+ * (C) Copyright (c) 2022 Amazon.com, Inc. or its affiliates. All rights reserved.
+ *
+ * This software is available to you under a choice of one of two
+ * licenses.  You may choose to be licensed under the terms of the GNU
+ * General Public License (GPL) Version 2, available from the file
+ * COPYING in the main directory of this source tree, or the
+ * BSD license below:
+ *
+ *     Redistribution and use in source and binary forms, with or
+ *     without modification, are permitted provided that the following
+ *     conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include "ofi_mr.h"
+#include "ofi_hmem.h"
+
+static int ipc_handle_monitor_subscribe(struct ofi_mem_monitor *monitor, const void *addr,
+					 size_t len, union ofi_mr_hmem_info *hmem_info)
+{
+	/* no-op */
+	return FI_SUCCESS;
+}
+
+static void ipc_handle_monitor_unsubscribe(struct ofi_mem_monitor *monitor, const void *addr,
+					    size_t len, union ofi_mr_hmem_info *hmem_info)
+{
+	/* no-op */
+}
+
+static bool ipc_handle_monitor_valid(struct ofi_mem_monitor *monitor, const void *addr,
+				      size_t len, union ofi_mr_hmem_info *hmem_info)
+{
+	/* no-op */
+	return true;
+}
+
+static int ipc_handle_monitor_start(struct ofi_mem_monitor *monitor)
+{
+	/* no-op */
+	return FI_SUCCESS;
+}
+
+void ipc_handle_monitor_stop(struct ofi_mem_monitor *monitor)
+{
+	/* no-op */
+}
+
+static struct ofi_mem_monitor ipc_handle_monitor = {
+	.init = ofi_monitor_init,
+	.cleanup = ofi_monitor_cleanup,
+	.start = ipc_handle_monitor_start,
+	.stop = ipc_handle_monitor_stop,
+	.subscribe = ipc_handle_monitor_subscribe,
+	.unsubscribe = ipc_handle_monitor_unsubscribe,
+	.valid = ipc_handle_monitor_valid,
```

Comment:
Do we have existing no-op options for the callbacks or can we set them to NULL?  If we don't have existing no-op options, move these to generic no-ops instead.

Reply:
We cannot set them to NULL because they are called explicitly in `ofi_mr_cache_search`, `util_mr_cache_create`, and `ofi_monitors_update`.

I did not find general no-op functions for `ofi_mem_monitor` yet. It seems all the other ifaces like cuda/ze/rocr all define the no-ops explicitly.

Reply:
We should define generic no-ops for these and update the other monitors to use them.

Reply:
I see, will do it then.

Reply:
Done. I was not able to update other monitors because they have a branching between no-op and ENOSYS according to `HAVE_CUDA/ZE/ROCR`.